### PR TITLE
[CBRD-22738] Refactoring regu_var.h & xasl_support.h

### PR DESCRIPTION
**Pull request by ribram**
http://jira.cubrid.org/browse/CBRD-22738

### Changes:

  1. Move all structures that are not related to regular variable from regu_var.h
  2. Move regu alloc/init functionality from xasl_support/parser_support to xasl_regu_alloc.

### Move aggregation/analytics from regu_var.h -
  1. Move XASL structures to xasl_aggregate.hpp & xasl_analytic.hpp
  2. Move query structures & functions from xasl.h/query_opfunc to query_aggregate & query_analytic
  3. Separate qdata_aggregate_interpolation & qdata_analytic_interpolation
  NOTE - some interpolation functions that are used by both aggregates and analytics remained in query_opfunc
  4. stx_init_analytic_type_unserialized_fields() -> analytic_list_node->init()
  5. buildlist_proc_node.agg_hash_context converted to pointer.

### Move predicate expressions from regu_var.h
  1. Move enums and structures to xasl_predicate.hpp
  2. Add m_ prefix to pred_expr union fields to avoid type name conflicts.
  3. Move re-eveluation structures and functions to query_reevaluation.

### Other regu_var.h
  1. Move heap_cache_attrinfo code to heap_attrinfo.h
  2. Replace UT_CLEAR_ATTR_DESCR() with attr_descr_node::reset()
  3. Move val_descr & xasl_state to query_executor
  4. Move qproc_db_value_list & val_list_node to xasl.h
  5. Move range structures & index_info to access_spec.hpp
  6. Removed REGU_VARIABLE_XASL
  7. Move METHOD_TYPE, METHOD_SIG & METHOD_SIG_LIST to method_def.hpp
  8. Move RANGE, KEY_VAL_RANGE, KEY_RANGE, KEY_INFO & INDX_INFO to access_spec.hpp.

### xasl_regu_alloc

1. Migrate all regu alloc functionality to xasl_regu_alloc from parser_support/xasl_support. Overloading regu_init changes how values are initialized. Existing types have been migrated by:
- use default constructor to initialize (instead of trivial initialization functions):
    QPROC_DB_VALUE_LIST/regu_dbvlist_alloc, ANALYTIC_EVAL_TYPE/regu_analytic_eval_alloc, VAL_LIST/regu_vallist_alloc, OUTPTR_LIST/regu_outlist_alloc, PRED_EXPR_WITH_CONTEXT/regu_pred_with_context_alloc, QFILE_SORTED_LIST_ID/regu_srlistid_alloc, HEAP_CACHE_ATTRINFO/regu_cache_attrinfo_alloc, ODKU_INFO/regu_odku_info_alloc, METHOD_SIG/regu_method_sig_alloc, METHOD_SIG_LIST/regu_method_sig_list_alloc, REGU_VALUE_LIST/regu_regu_value_list_alloc, REGU_VALUE_ITEM/regu_regu_value_item_alloc, FUNC_PRED/regu_func_pred_alloc
- init function kept as before:
    db_value/regu_dbval_alloc, aggregate_list_node/regu_agg_grbynum_alloc, regu_variable_node/regu_var_init, regu_variable_list_node/regu_varlist_alloc, pred_expr/regu_pred_alloc, analytic_list_node/regu_analytic_alloc, xasl_node/regu_xasl_node_alloc, access_spec_node/regu_spec_alloc, indx_info/regu_index_alloc, key_range/regu_keyrange_array_alloc, sort_list/regu_sort_alloc, OID/regu_oid_array_alloc, HFID/regu_hfid_array_alloc, qfile_list_id/regu_listid_alloc, tp_domain/regu_domain_db_alloc, upddel_class_info/regu_upddel_class_info_array_alloc, update_assignment/regu_update_assignment_array_alloc, selupd_list/regu_selupd_list_alloc
- merged init function with functionality from alloc function:
    arith_list_node/regu_arith_alloc, function_node/regu_func_alloc, aggregate_list_node/regu_agg_grbynum_alloc

2. Move functions to other places:
- regu_free_method_sig/regu_method_sig_list_init -> moved to query_method
- regu_free_domain/regu_cp_domain -> moved to class_object
- regu_free_listid -> replaced by cursor_free_self_list_id in cursor.h

3. Remove functions:
- regu_bytes_alloc, regu_string_alloc, regu_string_db_alloc, regu_string_ws_alloc, regu_strdup, regu_strcmp, regu_cp_listid
- regu_listid_db_alloc, regu_method_sig_db_alloc, regu_method_sig_list_db_alloc, regu_agg_alloc, regu_dbvallist_alloc

### Misc changes:
  1. Expose domain_init as tp_domain_init (used instead of regu_domain_db_alloc)
  2. Fix keywords INF_LE_ and INF_LT_

### Review requires special attention to:
- xasl_regu_alloc/parser_support/xasl_generation changes can have regressions.
- query_reevaluation
- buildlist_proc_node->proc.agg_hash_context

### CODE REVIEW

**Comment by ribram** [comment]
Patch requires manual shell & isolation testing before merging.

File: src/object/class_object.c [hunk 1]
```diff
@@ -8598,3 +8598,76 @@ classobj_change_constraint_status (DB_SEQ * properties, SM_CLASS_CONSTRAINT * co
   pr_clear_value (&new_status);
   return error;
 }
+
```
[the remark this answers is not in the document]
Comment (ribram) [reply src/object/class_object.c]: I will move the functions and SM_DOMAIN definition to schema_manager.

[the remark this answers is not in the document]
Comment (ribram) [reply src/object/class_object.c]: I moved only the functions.

File: src/query/xasl.h [hunk 2]
```diff
@@ -176,6 +188,25 @@ typedef enum
...
 } PROC_TYPE;
 
+typedef struct qproc_db_value_list *QPROC_DB_VALUE_LIST;	/* TODO */
+struct qproc_db_value_list
+{
+  QPROC_DB_VALUE_LIST next;
+  DB_VALUE *val;
+  TP_DOMAIN *dom;
+
+    qproc_db_value_list () = default;
```
Comment (razvanradulescu) [inline src/query/xasl.h]: This file seems to some indentation issues (mainly declaration of constructors of structs).

Comment (ribram) [reply src/query/xasl.h]: fixed the indentation issues.

**Pull request merged**